src/main/java/org/sagebionetworks/web/client/widget/team/JoinTeamWidget.java

```diff
@@ -365,10 +425,14 @@ public void sendJoinRequestStep3() {
 		synapseClient.requestMembership(authenticationController.getCurrentUserPrincipalId(), teamId, message, gwt.getHostPageBaseURL(), new AsyncCallback<Void>() {
 			@Override
 			public void onSuccess(Void result) {
-				String successJoinMessage = successMessage == null ? WidgetConstants.JOIN_TEAM_DEFAULT_SUCCESS_MESSAGE : successMessage;
-				String message = isAcceptingInvite ? successJoinMessage : "Request Sent";
-				view.showInfo(message, "");
-				refresh();
+				if (teamMembershipStatus.getCanJoin()) {
+					refresh();
+				} else {
+					view.showInfo("Request Sent", "");
```

Comment:
![image](https://cloud.githubusercontent.com/assets/8346437/9453150/4775a61c-4a6d-11e5-8a96-e5ffa5509c38.png)

From a non-simple join, where you send a request.